airbyte-integrations/connectors/source-s3/source_s3/v4/legacy_config_transformer.py

```diff
@@ -0,0 +1,47 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+from source_s3.source import SourceS3Spec
+from typing import Mapping, Any, List
+
+
+class LegacyConfigTransformer:
+    """
+    Class that takes in S3 source configs in the legacy format and transforms them into
+    configs that can be used by the new S3 source built with the file-based CDK.
+    """
+
+    @classmethod
+    def convert(cls, legacy_config: SourceS3Spec) -> Mapping[str, Any]:
+        transformed_config = {
+            "bucket": legacy_config.provider.bucket,
+            "streams": [
+                {
+                    "name": legacy_config.dataset,
+                    "file_type": legacy_config.format.filetype,
+                    "globs": cls.create_globs(legacy_config.path_pattern, legacy_config.provider.path_prefix),
+                    "validation_policy": "emit_record",
```

Comment:
```suggestion
                    "validation_policy": "Emit Record",
```

Reply:
yep fixed that after I merged in the latest master